x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Analyzer.java

```diff
@@ -771,9 +771,9 @@ protected LogicalPlan resolve(LogicalPlan plan, Map<String, List<Function>> seen
                         return uf;
                     }
 
-                    String normalizedName = functionRegistry.concreteFunctionName(name);
+                    name = functionRegistry.concreteFunctionName(name);
```

Comment:
I am not convinced that this change combined with https://github.com/elastic/elasticsearch/pull/34137/files#diff-bb6b97c81fc9a1971733dc1d3b90be73R229 is the right approach. LGTM if @costin is also ok with it.

Reply:
The 2 function calls must remain discrete since in-between we need to do the resolution through the "seen" functions. I choose to remove the new string variable and re-assign to name as the `concreteFunctionName()` doesn't guarantee that it returns the actual concrete function name, it's just an attempt, so I didn't like to use a variable with a name like `concreteName`. I'd also like to rename the `concreteFunctionName()` to something like `tryResolveFromAliases`.

Do you maybe have a good suggestion for a variable name here?

Reply:
I'm not sure I follow - why not use `functionName` or simply `resolvePotentialAlias`/`resolveAlias`/`getName`?